components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -200,12 +210,28 @@ private void handleRequest(HttpServletRequest req, HttpServletResponse resp, boo
             if(log.isDebugEnabled()){
                 log.debug("Error occurred while handling SAML2 SSO request", e);
             }
-            sendNotification(SAMLSSOConstants.Notification.EXCEPTION_STATUS,
-                    SAMLSSOConstants.Notification.EXCEPTION_MESSAGE, req, resp);
+            String errorResp = null;
+            try {
+                errorResp = SAMLSSOUtil.buildErrorResponse(
+                        SAMLSSOConstants.StatusCodes.IDENTITY_PROVIDER_ERROR,
+                        "Error occurred while handling SAML2 SSO request");
+            } catch (IdentityException e1) {
+                log.error("Error while building SAML response");
+            }
+            sendNotification(errorResp, SAMLSSOConstants.Notification.EXCEPTION_STATUS,
+                    SAMLSSOConstants.Notification.EXCEPTION_MESSAGE, null, req, resp);
         } catch (IdentityException e) {
             log.error("Error when processing the authentication request!", e);
-            sendNotification(SAMLSSOConstants.Notification.EXCEPTION_STATUS,
-                    SAMLSSOConstants.Notification.EXCEPTION_MESSAGE, req, resp);
+            String errorResp = null;
+            try {
+                errorResp = SAMLSSOUtil.buildErrorResponse(
+                        SAMLSSOConstants.StatusCodes.IDENTITY_PROVIDER_ERROR,
+                        "Error when processing the authentication request");
+            } catch (IdentityException e1) {
+                log.error("Error while building SAML response");
```

Comment:
Exception is logged in later commits. https://github.com/wso2/carbon-identity/pull/399/files#diff-8c888107ac851d00c05903c8163d21b7R232